lib/rbi/rewriters/attr_to_methods.rb

```diff
@@ -0,0 +1,150 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RBI
+  module Rewriters
+    class AttrToMethods < Visitor
+      extend T::Sig
+
+      sig { override.params(node: T.nilable(Node)).void }
+      def visit(node)
+        return unless node
+
+        case node
+        when Tree
+          node.nodes.dup.each do |child|
+            visit(child)
+            next unless (attr = child).is_a?(Attr)
+
+            new_methods = convert_to_methods(attr)
+
+            replace(child, with: new_methods)
+          end
```

Comment:
Can't we simplify this actually?

```suggestion
        return unless node

        case node
        when Attr
          new_methods = convert_to_methods(attr)
          replace(child, with: new_methods)
        when Tree
          visit_all(node.nodes)
        end
```

Reply:
You could even hoist up the logic from `convert_to_methods`?